x-pack/legacy/plugins/rollup/__jest__/client_integration/job_create_metrics.test.js

```diff
@@ -247,12 +250,157 @@ describe('Create Rollup Job, step 5: Metrics', () => {
+        selectAllAvgCheckbox.first().simulate('change', { checked: true });
+
+        const rows = getFieldListTableRows();
+
+        rows
+          .filter((row) => {
+            const [, metricTypeCol ] = row.columns;
+            return metricTypeCol.value === 'numeric';
+          })
+          .forEach((row, idx) => {
+            const fieldChooser = getFieldChooserColumnForRow(idx);
+            fieldChooser.reactWrapper.find('input').forEach(input => {
+              const props = input.props();
+              if (props['data-test-subj'].endsWith('avg')) {
+                expect(props.checked).toBe(true);
+              } else {
+                expect(props.checked).toBe(false);
+              }
+            });
+          });
+      });
+
+      it('should deselect all of the metric types across rows (column-wise)',  () => {
+        const selectAllAvgCheckbox = find('rollupJobMetricsSelectAllCheckbox-avg');
+        selectAllAvgCheckbox.last().simulate('change', { checked: true });
+        selectAllAvgCheckbox.last().simulate('change', { checked: false });
+
+        const rows = getFieldListTableRows();
+
+        rows.forEach((row, idx) => {
+          const [, metricTypeCol ] = row.columns;
+          if (metricTypeCol.value !== 'numeric') {
+            return;
+          }
+          const fieldChooser = getFieldChooserColumnForRow(idx);
+          fieldChooser.reactWrapper.find('input').forEach(input => {
+            expect(input.props().checked).toBe(false);
+          });
+        });
+      });
+
+      it('should correctly select across rows and columns',  async () => {
+        /**
+         * Tricky test case where we want to determine that the column-wise and row-wise
+         * selections are interacting correctly with each other.
+         *
+         * We will select avg (numeric) and max (numeric + date) to ensure that we are
+         * testing across metrics types too. The plan is:
+         *
+         * 1. Select all avg column-wise
+         * 2. Select all max column-wise
+         * 3. Select and deselect row-wise the first numeric metric row
+         * 4. Expect the avg and max select all checkboxes to be unchecked
+         * 5. Select all on the last date metric row-wise
+         * 6. Select then deselect all max column-wise
+         * 7. Expect everything but all and max to be selected on the last date metric row
+         *
+         * Let's a go!
+         */
+
+        // 1.
+        find('rollupJobMetricsSelectAllCheckbox-avg').first().simulate('change', { checked: true });
+        // 2.
+        find('rollupJobMetricsSelectAllCheckbox-max').first().simulate('change', { checked: true });
+
+        const selectAllCheckbox = getSelectAllInputForRow(0);
+
+        // 3.
+        // Select All
+        selectAllCheckbox.simulate('change', { checked: true });
+        // Deselect All
+        selectAllCheckbox.simulate('change', { checked: false });
```

Comment:
similar question as above - are these two steps needed? by default, will the checkboxes not be unselected?

Reply:
In this case we want to see that selecting some checkboxes column wise, then selecting the row-wise and then de-selecting them row-wise doesn't leave us in a weird state where some checkboxes (esp. All) are still selected for that row. Will add comment though :).